src/arcade/potts/env/location/PottsLocation.java

```diff
@@ -627,141 +635,35 @@ Location separateVoxels(ArrayList<Voxel> voxelsA, ArrayList<Voxel> voxelsB,
-                    break;
-                case POSITIVE_XY:
-                    if (voxel.x - center.x > voxel.y - center.y) {
-                        voxelsA.add(voxel);
-                    } else if (voxel.x - center.x < voxel.y - center.y) {
-                        voxelsB.add(voxel);
-                    } else {
-                        if (random.nextDouble() > 0.5) {
-                            voxelsA.add(voxel);
-                        } else {
-                            voxelsB.add(voxel);
-                        }
-                    }
-                    break;
-                case NEGATIVE_YZ:
-                    if (voxel.y - center.y > center.z - voxel.z) {
-                        voxelsA.add(voxel);
-                    } else if (voxel.y - center.y < center.z - voxel.z) {
-                        voxelsB.add(voxel);
-                    } else {
-                        if (random.nextDouble() > 0.5) {
-                            voxelsA.add(voxel);
-                        } else {
-                            voxelsB.add(voxel);
-                        }
-                    }
-                    break;
-                case POSITIVE_YZ:
-                    if (voxel.y - center.y > voxel.z - center.z) {
-                        voxelsA.add(voxel);
-                    } else if (voxel.y - center.y < voxel.z - center.z) {
-                        voxelsB.add(voxel);
-                    } else {
-                        if (random.nextDouble() > 0.5) {
-                            voxelsA.add(voxel);
-                        } else {
-                            voxelsB.add(voxel);
-                        }
-                    }
-                    break;
-                case NEGATIVE_ZX:
-                    if (voxel.z - center.z > center.x - voxel.x) {
-                        voxelsA.add(voxel);
-                    } else if (voxel.z - center.z < center.x - voxel.x) {
-                        voxelsB.add(voxel);
-                    } else {
-                        if (random.nextDouble() > 0.5) {
-                            voxelsA.add(voxel);
-                        } else {
-                            voxelsB.add(voxel);
-                        }
-                    }
-                    break;
-                case POSITIVE_ZX:
-                    if (voxel.z - center.z > voxel.x - center.x) {
-                        voxelsA.add(voxel);
-                    } else if (voxel.z - center.z < voxel.x - center.x) {
-                        voxelsB.add(voxel);
-                    } else {
-                        if (random.nextDouble() > 0.5) {
-                            voxelsA.add(voxel);
-                        } else {
-                            voxelsB.add(voxel);
-                        }
-                    }
-                    break;
-                default:
-                    break;
+            if (plane.distanceToPlane(new Int3D(voxel.x, voxel.y, voxel.z)) < 0) {
```

Comment:
Move the distance calculation outside of the condition so you only need to call it once.